packages/suite/src/components/suite/NumberInput.tsx

```diff
@@ -104,7 +101,12 @@ export const NumberInput = ({
         defaultValue,
     });
 
+    const locale = useSelector(state => state.suite.settings.language);
+    const [pressedKey, setPressedKey] = useState('');
     const [displayValue, setDisplayValue] = useState(localizeNumber(value || '', locale));
+    const initialHistoryValue = value ? [value] : [''];
```

Comment:
Will this work correctly if `value` is `0`? Same for the line above. I think a better approach might be to provide a default value for `value` on line 95 (`value = ''`) and remove the fallbacks from lines 106 and 107.

Reply:
f52f6b73d